django/template/context.py

```diff
@@ -251,7 +251,15 @@ def bind_template(self, template):
         processors = template.engine.template_context_processors + self._processors
         updates = {}
         for processor in processors:
-            updates.update(processor(self.request))
+            context = processor(self.request)
+            try:
+                updates.update(context)
+            except TypeError as e:
+                raise TypeError(
+                    f"Context processor {processor.__qualname__} didn't return a "
+                    "dictionary."
+                ) from e
```

Comment:
`as e` / `from e` are unnecessary, since that’s the default behaviour. But oh well.

Reply:
@adamchainz You're probably more familiar with chaining than me but isn't the implicit version slightly different? 🤔  At least that's how I'm reading it in the docs:

https://docs.python.org/3/tutorial/errors.html#exception-chaining

More specifically: https://docs.python.org/3/reference/simple_stmts.html#raise

Using `from` attaches the parent to the `__cause__` as opposed to the `__context__` with the implicit variety 🤔

Reply:
I checked this too. Without `from e` the message that's printed says, "During handling of the above exception, another exception occurred:" but with it "This exception is the direct cause of the following exception".

Reply:
Right y'all are. My bad!